purchase_requisition_from_rfq/models/purchase_requisition.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = "purchase.requisition"
+
+    purchase_ids = fields.One2many(
+        default=lambda self: self._context.get('active_ids', []),
+    )
+    line_ids = fields.One2many(
+        default=lambda self: self._get_line(),
+    )
+    from_rfq = fields.Boolean(
+        default=False,
+        readonly=True,
+        help="True, if this purchase agreement is created from RFQs",
+    )
+
+    @api.multi
+    def _get_line(self):
+        if self._context.get('active_ids'):
+            order = orders = self.env['purchase.order'].browse(
+                self._context.get('active_ids', []))
+            if orders.mapped('requisition_id'):
+                raise ValidationError(_(
+                    'Order: %s, was used in some agreement!') % ', '.join(
+                    orders.filtered(lambda l: l.requisition_id).mapped('name')))
+            po_dict = {}
+            len_max = 0
```

Comment:
????

Reply:
I need to use len_max for keep max number of line in PO.
But I forgot this code `len_max = len(po_dict[order.id])`, now I add it.